src/monitor-plan/monitor-plan.service.ts

```diff
@@ -312,21 +314,21 @@ export class MonitorPlanService {
       promises.push(
         new Promise((resolve, _reject) => {
           (async () => {
-            const quals = await this.qualificationRepository.find({
+            const quals = await useSlaveRepository(this.dataSource, MonitorQualificationRepository, async (repository) => repository.find({
               where: { locationId: In(locationIds) },
-            });
+            }));
 
             if (quals.length !== 0) {
               const qualIds = quals.map(i => i.id);
-              const q1 = this.leeQualificationRepository.find({
+              const q1 = await useSlaveRepository(this.dataSource, LEEQualificationRepository, async (repository) => repository.find({
                 where: { qualificationId: In(qualIds) },
-              });
-              const q2 = this.lmeQualificationRepository.find({
+              }));
+              const q2 = await useSlaveRepository(this.dataSource, LMEQualificationRepository, async (repository) => repository.find({
```

Comment:
This should also not use await, since the promise is resolved in the Promise.all() below.
```suggestion
              const q2 = useSlaveRepository(this.dataSource, LMEQualificationRepository, async (repository) => repository.find({
```

Reply:
updated this in latest commit